pkg/application/service/factory/service_factory.go

```diff
@@ -0,0 +1,78 @@
+package factory
+
+import (
+	"github.com/codeready-toolchain/registration-service/pkg/application/service"
+	"github.com/codeready-toolchain/registration-service/pkg/kubeclient"
+	signup_service "github.com/codeready-toolchain/registration-service/pkg/signup/service"
+	verification_service "github.com/codeready-toolchain/registration-service/pkg/verification/service"
+	"github.com/prometheus/common/log"
+
+	servicecontext "github.com/codeready-toolchain/registration-service/pkg/application/service/context"
+	"github.com/codeready-toolchain/registration-service/pkg/configuration"
+)
+
+type serviceContextImpl struct {
+	kubeClient kubeclient.CRTClient
+	services   service.Services
+}
+
+func NewServiceContext(kubeClient kubeclient.CRTClient, config configuration.Configuration, options ...Option) servicecontext.ServiceContext {
+	ctx := &serviceContextImpl{kubeClient: kubeClient}
+	var sc servicecontext.ServiceContext
+	sc = ctx
```

Comment:
hmm it's a bit strange to create the ctx variable only to re-assign it to a servicecontext

Reply:
I agree, but this quirk is required for this code to work.

Reply:
I've done some refactoring here which has allowed me to remove this code altogether.